### ISSUE TITLE
Match / add Highcharts API to the global options (specifically setOptions)

### ISSUE THREAD

**Issue by JohnMica** [richardeschloss/nuxt-highcharts#14]
ideally the  global options could have more options added / be updated to reflect the Highcharts API.

```
// module options
highcharts : {
        chart: {
            style: {
              fontFamily: '"Roboto", sans-serif'
            },
            zoomType: 'xy'
         },
        chartOptions: {
          legend: {
            layout: 'horizontal',
            align: 'center',
            verticalAlign: 'bottom'
          }
        },
        credits: { enabled: false },
        exporting: { enabled: false },
        boost: {
          enabled: true,
          useGPUTranslations: true,
          usePreallocated: true,
          allowForce: true
          // seriesThreshold: 20
        }
...
```
not sure this could match you ideas for future development ?

**Comment by richardeschloss** [comment]
Hmm, I actually think all those options sibling to "chartOptions" actually belong in "chartOptions" (i.e., chart, credits, exporting, boost are all "chartOptions"). I think the easiest thing for users to do would be to take advantage of the [Highcharts types](https://www.highcharts.com/docs/advanced-chart-features/highcharts-typescript-declarations) that ship with highcharts ([ChartOptions](https://github.com/highcharts/highcharts/blob/master/ts/Core/Options.ts) so that they get the intellisense as they type the chartOptions. While chartOptions are respected in the nuxt.config, I think a more flexible design could be one that has those chartOptions maintained / keyed in a separate file that individual components lazily consume from. 

To get the intellisense, it would be done like (I think, I haven't tested it):

```js
// @type {import('highcharts').ChartOptions}  // <-- intellisense should let you pick the correct imported type here
const chartOptions = { 
  // type "e" and you should see "exporting" show up in the list
}

// ... use chartOptions later on...
```

I think the Intellisense may be easier to use if the chartOptions are in a typescript file and the tsconfig.json is set up correctly (however, I'm not of a typescript expert, so I don't know for sure)


---
API reference here: https://api.highcharts.com/highcharts/

**Comment by richardeschloss** [comment]
I think what may be more appropriate, in the context of this issue, would be to support the options that would be passed to "setOptions":

```js
highcharts: {
  setOptions: { // <-- module should consume this
    // pass these options to the Highcharts.setOptions method
  }
  chartOptions: { /* already supported */ }
}
```

Currently, the setOptions are "global" or "lang", but it's possible those can change, so for wiring this module to it's api, it'll be easiest to have a "setOptions" prop that gets consumed. This too can be used with their types and intellisense to aid the user.

**Comment by richardeschloss** [comment]
Added the label "good first issue" since I think it could be straightforward to add:

plugin.js consumes the pluginOptions and then if pluginOptions.setOptions is truthy, we simply pass along that to Highcharts.setOptions. It may be easier to test this if this code is outside of the plugin.js file.

**Comment by richardeschloss** [comment]
Working code and intellisense example(s) pushed to [feat/setOptions branch](https://github.com/richardeschloss/nuxt-highcharts/tree/feat/setOptions). Just have to update the unit tests.

**Comment by richardeschloss** [comment]
Merged #18, released 1.0.5

**Comment by JohnMica** [comment]
thanks for this @richardeschloss. will start to test it today and let you know what and how
